packages/core/src/StateNode.ts

```diff
@@ -1322,6 +1324,10 @@ class StateNode<
     // Preserve original history after raised events
     maybeNextState.history = history;
 
+    maybeNextState.tags = new Set(
+      flatten(maybeNextState.configuration.map((sn) => sn.tags ?? []))
```

Comment:
I think that it would be preferable to normalize `tags` to an empty array in the constructor, so we wouldn't have to deal with this here.

Reply:
Done